quartz/plugins/transformers/frontmatter.ts

```diff
@@ -8,12 +8,12 @@ import { QuartzPluginData } from "../vfile"
 import { i18n } from "../../i18n"
 
 export interface Options {
-  delims: string | string[]
+  delimiters: string | [string, string]
   language: "yaml" | "toml"
 }
 
 const defaultOptions: Options = {
-  delims: "---",
+  delimiters: "---",
```

Comment:
This change is breaking

Reply:
What is the version of gray-matte ryou are using?

Reply:
The plugin passes through the options to the backing library (gray-matter) without any checks. Since the method signature there is supposed to be `delimiters` since 2019, the current state is depending on the library to continue supporting the long deprecated format. As soon as it doesn't any more, this will break silently.

Since the whole options object is passed through, this change will *not* affect users who have set `delims` themselves as the `delims` property will still be passed through. In that case, both the user set `delims` and the default `delimiters` will be passed to the underlying library. But that is fine, because there `delims` is prioritized over `delimiters` (see the linked issue and my comment there).

So **with** this change everything will still work but the users will get a TS error, telling them to fix this, which is good. **Without** this change it will some day break silently.

The other option would be to keep `delims` in the user facing options and map this to `delimiters` before passing it to the library.

Reply:
<img width="748" alt="Screenshot 2024-02-18 at 13 50 20" src="https://github.com/jackyzha0/quartz/assets/29749331/c1fca221-4d1c-472e-9889-fc27d959cbdb">
They have backward compatible changes here.

I will close this until it breaks.

Reply:
I would prefer for this to update the interface type, so once we upgrade gray-matter it will raise error then.

Reply:
The interface of gray-matter already only exposes like this:

```
  interface GrayMatterOption<
    I extends Input,
    O extends GrayMatterOption<I, O>
  > {
  ...
    language?: string
    delimiters?: string | [string, string]
  }
```

The backward compatibility is done way later. If it is removed there some day, you will not see it breaking (on interface level).

Reply:
They haven't updated the types yet, so if you changed to delimiters it will still fail.

Reply:
> They haven't updated the types yet, so if you changed to delimiters it will still fail.

The quote from my previous comment is from `node_modules/gray-matter/gray-matter.d.ts`. Those only support `delimiters`. Do we talk about the same thing here?